pkg/controller/acmechallenges/sync.go

```diff
@@ -92,8 +93,19 @@ func (c *controller) Sync(ctx context.Context, chOriginal *cmacme.Challenge) (er
 	// This finalizer ensures that the challenge is not garbage collected before
 	// cert-manager has a chance to clean up resources created for the
 	// challenge.
+	//
+	// API Transition
+	// -- we are adding cmacme.ACMELegacyFinalizer today.
+	// -- in a future version we intend to add cmacme.ACMEDomainQualifiedFinalizer instead.
+	// -- if cmacme.ACMEDomainQualifiedFinalizer is present, we don't need to add a finalizer.
```

Comment:
In both cases ( cmacme.ACMELegacyFinalizer and cmacme.ACMEDomainQualifiedFinalizer ) we do not need to add a finalizer.